main.py

```diff
@@ -481,6 +484,92 @@ def delete_interface_metadata(self, request: Request) -> JSONResponse:
         self.notify_metadata_changes(interface, 'removed')
         return JSONResponse("Operation successful")
 
+    @staticmethod
+    def _get_tag_type(content):
+        tag_type = content.get("tag_type", '1')
+        if tag_type != str(TAGType.VLAN.value):
+            detail = f"The TAG type {tag_type} is not allowed."
+            raise HTTPException(400, detail=detail)
+        return tag_type
+
+    @staticmethod
+    def check_non_range(tag_range):
+        """Change integer or singular interger list to
+        list[int, int] when necessary"""
+        if isinstance(tag_range, int):
+            tag_range = [tag_range] * 2
+        elif len(tag_range) == 1:
+            tag_range = [tag_range[0]] * 2
+        return tag_range
+
+    def _get_tag_ranges(self, request: Request):
+        """Get tag_ranges and check validity:
+        - It should be ordered
+        - Not unnecessary partition (eg. [[10,20],[20,30]])
+        - Singular intergers are changed to ranges (eg. [10] to [[10, 10]])
+        The ranges are understood as [inclusive, inclusive]"""
+        content_type_json_or_415(request)
+        content = get_json_or_400(request, self.controller.loop)
+        ranges: list[Union[int, list[int]]] = content["tag_ranges"]
+        tag_type = self._get_tag_type(content)
+        ranges[0] = self.check_non_range(ranges[0])
+        for i in range(1, len(ranges)):
+            ranges[i] = self.check_non_range(ranges[i])
+            if ranges[i][0] > ranges[i][1]:
+                detail = f"The range {ranges[i]} is not ordered"
+                raise HTTPException(400, detail=detail)
+            if ranges[i-1][1] > ranges[i][0]:
+                detail = f"tag_ranges is not ordered. {ranges[i-1][1]}"\
+                         f" is higher than {ranges[i][0]}"
+                raise HTTPException(400, detail=detail)
+            if ranges[i-1][1] == ranges[i][0] - 1:
+                detail = f"tag_ranges has an unnecessary partition. "\
+                         f"{ranges[i-1][1]} is before to {ranges[i][0]}"
+                raise HTTPException(400, detail=detail)
+            if ranges[i-1][1] == ranges[i][0]:
+                detail = f"tag_ranges has repetition. {ranges[i-1]}"\
+                         f" have same values as {ranges[i]}"
+                raise HTTPException(400, detail=detail)
+        if ranges[-1][1] > 4095:
+            detail = "Maximum value for tag_ranges is 4095"
+            raise HTTPException(400, detail=detail)
+        if ranges[0][0] < 1:
+            detail = "Minimum value for tag_ranges is 1"
+            raise HTTPException(400, detail=detail)
+        return ranges, tag_type
+
+    @rest('v3/interfaces/{interface_id}/tag_ranges', methods=['POST'])
+    @validate_openapi(spec)
+    def add_tag_range(self, request: Request) -> JSONResponse:
```

Comment:
If you could rename the add with set for clarity that'd be great, like you've done for the openapi spec, since this is actually setting and not adding.

Reply:
Changed to `set_tag_range` and updated docstrings